samples/java/chatapp/src/main/java/com/webpubsub/tutorial/App.java

```diff
@@ -41,29 +45,64 @@ public static void main(String[] args) {
             option.setUserId(id);
             WebPubSubClientAccessToken token = service.getClientAccessToken(option);
             ctx.contentType("application/json");
-            String response = String.format("{\"url\":\"%s\"}", token.getUrl());
+            Gson gson = new Gson();
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.addProperty("url", token.getUrl());
+            String response = gson.toJson(jsonObject);
             ctx.result(response);
             return;
         });
 
-        
-        // validation: https://azure.github.io/azure-webpubsub/references/protocol-cloudevents#validation
+        // validation:
+        // https://learn.microsoft.com/azure/azure-web-pubsub/reference-cloud-events#protection
         app.options("/eventhandler", ctx -> {
             ctx.header("WebHook-Allowed-Origin", "*");
         });
 
-        // handle events: https://azure.github.io/azure-webpubsub/references/protocol-cloudevents#events
+        // handle events:
+        // https://learn.microsoft.com/azure/azure-web-pubsub/reference-cloud-events#events
         app.post("/eventhandler", ctx -> {
             String event = ctx.header("ce-type");
-            if ("azure.webpubsub.sys.connected".equals(event)) {
+            // handle connect event when lazy auth is enabled as described in 
+            if ("azure.webpubsub.sys.connect".equals(event)) {
+                String body = ctx.body();
+                System.out.println("Reading from request body...");
+                Gson gson = new Gson();
+                JsonObject requestBody = gson.fromJson(body, JsonObject.class); // Parse JSON request body
+                JsonObject query = requestBody.getAsJsonObject("query");
+                if (query != null) {
+                    System.out.println("Reading from request body query:" + query.toString());
+                    JsonElement idElement = query.get("id");
+                    if (idElement != null) {
+                        JsonArray idInQuery = query.get("id").getAsJsonArray();
+                        if (idInQuery != null && idInQuery.size() > 0) {
+                            String id = idInQuery.get(0).getAsString();
+                            ctx.contentType("application/json");
+                            Gson response = new Gson();
+                            JsonObject jsonObject = new JsonObject();
+                            jsonObject.addProperty("userId", id);
+                            ctx.result(response.toJson(jsonObject));
+                            return;
```

Comment:
Is it by-design to skip set status in this condition?

Reply:
ctx.result set status to 200